### PR TITLE
feat: render task dependency graph with G6

### PR DESCRIPTION
**Pull request by tacascer**
## Summary
- Migrates the task dependency graph from D3/D3-DAG SVG rendering to AntV G6.
- Uses G6's `antv-dagre` layered DAG layout as the supported Sugiyama-style layout for task dependencies.
- Adds Angular/Bazel wiring for G6 dependencies and WASM runtime assets.

## Verification
- `bazel run gazelle`
- `bazel run //tools/format`
- `aspect test //angular/projects/build-scan-web:test`
- `aspect build //angular/projects/build-scan-web`
- `aspect lint //...`
- `aspect test //...`
- `aspect build //...`
- Agent-browser smoke test on local scan `3aeba0b1-fa7d-4ca4-9c62-d4e1e3ab1366`: graph container rendered 4 canvas layers at 1059x384 with 45 nodes / 127 edges and no console/page errors.

## Notes
- Screenshot from local browser verification: `/tmp/task-dependency-g6-rendered.png`.
- G6 does not expose a separate public `sugiyama` layout key; `antv-dagre` is the supported layered DAG/Sugiyama-style layout path.

### CODE REVIEW

File: angular/projects/build-scan-web/src/app/scans/task-dependency-graph/task-dependency-graph.component.ts [hunk 1]
```diff
@@ -487,164 +389,69 @@ export class TaskDependencyGraphComponent {
...
-      nodes: positionedNodes,
-      edges,
-      width,
-      height,
-      zoomKey: buildZoomKey({ nodes: positionedNodes, edges, width, height }),
+      data: { nodes, edges },
+      key: buildGraphDataKey(graph),
+      nodeIds: graph.nodes.map((node) => node.id),
+      edgeIds: graph.edges.map((edge) => edgeKey(edge)),
     };
```
[the remark this answers is not in the document]
Comment (tacascer) [reply angular/projects/build-scan-web/src/app/scans/task-dependency-graph/task-dependency-graph.component.ts:419]: Fixed in a8ba7d7 by sorting dependency edges deterministically by sourceId, then targetId, before computing graph data/key material. I also added a regression test that verifies edge order is stable even when task input order differs.
